sklearn/decomposition/incremental_pca.py

```diff
@@ -61,6 +61,11 @@ class IncrementalPCA(_BasePCA):
         is inferred from the data and set to ``5 * n_features``, to provide a
         balance between approximation accuracy and memory consumption.
 
+    allow_smaller_final_batch : bool, (default=True)
```

Comment:
I don't think we need to add another function. I think we need to allow for small batches in partial_fit in general. Maybe we need to check the total number used in fitting.